### PR TITLE
add cuboid annotations to client

### PR DESCRIPTION
**Pull request by drew-kaul**
This PR adds client side support for CuboidAnnotations and CuboidPredictions. The endpoint for uploading predictions and annotations already supports cuboids, so uploading cuboid annotations and predictions to a dataset from the Python client should work now.

Edit: Backend changes for /exportForTraining have landed

Changed DATASET_ITEM_ID_KEY to ITEM_ID_KEY in annotation/prediction classes since backend expects the field "item_id" to be populated

### CODE REVIEW

File: nucleus/dataset.py [hunk 1]
```diff
@@ -168,9 +167,6 @@ def annotate(
             "ignored_items": int,
         }
         """
-        if any((isinstance(ann, CuboidAnnotation) for ann in annotations)):
```
Comment (ardila) [inline nucleus/dataset.py:171]: still need this check in dataset.annotate since you can't upload them yet

**Review by ardila — APPROVED** [review]
LGTM but even I'm not clear on what is supported or not.

Can you at least clarify in the PR description what is expected to work and what isn't?

File: nucleus/dataset.py (hunk 1 above)
Comment (drew-kaul) [reply nucleus/dataset.py:171]: cuboids can be uploaded, the backend changes are already there. here's the tests for that:
[tests](https://github.com/scaleapi/scaleapi/blob/master/test/nucleus/testEndpoints.ts#L786)

Comment (ardila) [reply nucleus/dataset.py:171]: should add test coverage for this here then to make sure the format generated by the client is compatible with the backend

**Pull request merged**